phpunit/functional/Glpi/Form/Destination/CommonITILField/RequestSourceFieldTest.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace tests\units\Glpi\Form\Destination\CommonITILField;
+
+use DbTestCase;
+use Glpi\Form\AnswersHandler\AnswersHandler;
+use Glpi\Form\Destination\CommonITILField\RequestSourceFieldConfig;
+use Glpi\Form\Destination\CommonITILField\RequestSourceFieldStrategy;
+use Glpi\Form\Destination\FormDestinationTicket;
+use Glpi\Form\Form;
+use Glpi\Tests\FormBuilder;
+use Glpi\Tests\FormTesterTrait;
+use RequestType;
+use Ticket;
+use TicketTemplate;
+use TicketTemplatePredefinedField;
+
+final class RequestSourceFieldTest extends DbTestCase
+{
+    use FormTesterTrait;
+
+    public function testDefaultSourceWithPredefinedField(): void
+    {
+
+        $source = RequestType::getDefault('helpdesk');
+
+        $default_template = (new Ticket())->getITILTemplateToUse(
+            entities_id: $_SESSION["glpiactive_entity"]
+        );
+        $this->createItem(
+            TicketTemplatePredefinedField::class,
+            [
+                'tickettemplates_id' => $default_template->getID(),
+                'num' => 9,
+                'value' => $source,
+            ]
+        );
+
+        $created_ticket = $this->checkRequestSourceFieldConfiguration(
+            form: $this->createAndGetFormWithTicketDestination(),
+            config: new RequestSourceFieldConfig(
+                strategy: RequestSourceFieldStrategy::FROM_TEMPLATE,
+            ),
+            expected_source_id: $source
```

Comment:
Tests look good but the source is checked twice (once in `checkRequestSourceFieldConfiguration` and one after it).
I guess we can keep only one check, I prefer the second one.

```suggestion
```